ui/src/lib/reactFlow/nodeTypes/GeneralNode/hooks.ts

```diff
@@ -34,6 +44,17 @@ export default ({ data, type }: { data: NodeData; type: string }) => {
   const [borderColor, selectedColor, selectedBackgroundColor] =
     getNodeColors(type);
 
+  const handleCollapsedToggle = useCallback(
+    (collapsed: boolean) => {
+      const yNodes = currentYWorkflow?.get("nodes") as YNodesMap | undefined;
+      const yNode = yNodes?.get(nodeId);
+      if (!yNode) return;
+      const yData = yNode?.get("data") as Y.Map<YNodeValue>;
+      yData?.set("isCollapsed", collapsed);
+    },
+    [currentYWorkflow, nodeId],
+  );
+
```

Comment:
Ah since this isn't wrapped in the undoTracker its not undo-able. If possible, please wrap here so it is treated like all other updates